### PR TITLE
Feature/kick command

### PR DESCRIPTION
**Pull request by Ikinon**
# Pull request

 - [x] I have read and agreed to the [CODE OF CONDUCT](https://github.com/CascadeBot/CascadeBot/blob/master/CODE_OF_CONDUCT.md).
 - [x] What I'm implementing was assigned to me or was not being worked on and is a somewhat requested feature. For reference see our [trello](https://trello.com/b/gUQehy4l/bot).
 - [ ] I have tested all of my changes.
 - [ ] I have followed the style of the rest of the project.
 

## Feature
Kick Command

### CODE REVIEW

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.cascadebot.cascadebot.commandmeta.Module;
+import com.cascadebot.cascadebot.permissions.CascadePermission;
+import com.cascadebot.cascadebot.utils.DiscordUtils;
+import net.dv8tion.jda.core.entities.Member;
+
+public class KickCommand implements ICommandMain {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        Member Target = DiscordUtils.getMember(context.getArg(0),context.getGuild());
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java]: camelCase

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        String User = DiscordUtils.getMember(context.getArg(0),context.getGuild()).getUser().getAsTag();
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java]: 1. camelCase
2. DiscordUtils.getMember can return null.

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+    @Override
+    public String command() {
+        return "kick";
+    }
+
+    @Override
+    public CascadePermission getPermission() {
+        return CascadePermission.of("Kick Command", "Kick", true);
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java]:
```suggestion
        return CascadePermission.of("Kick Command", "kick", true);
```

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import com.cascadebot.cascadebot.commandmeta.Module;
+import com.cascadebot.cascadebot.permissions.CascadePermission;
+import com.cascadebot.cascadebot.utils.DiscordUtils;
+import net.dv8tion.jda.core.entities.Member;
+
+public class KickCommand implements ICommandMain {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        Member TargetMember = DiscordUtils.getMember(context.getArg(0),context.getGuild());
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java]: 1. camelCase
2. DiscordUtils.getMember can return null
3. you don't know for sure that the user specified another user

Comment (weeryan17) [reply src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java]: Alos you sould be combining all the args for getting the user. we have a util method for that

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+public class KickCommand implements ICommandMain {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        Member targetMember = DiscordUtils.getMember(context.getArg(0),context.getGuild());
+        if (targetMember == null){
+            context.replyDanger("Could not find that user");
+            return;
+        }
+        if (targetMember != null) {
```
Comment (weeryan17) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java]: you can just use an else statment

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        if (targetMember == null) {
+            context.replyDanger("Could not find that user");
+        } else {
+            try {
+                context.getGuild().getController().kick(targetMember).queue();
+                context.replyInfo("User: " + targetMember.getUser().getAsTag() + " has been kicked");
+            } catch (InsufficientPermissionException e) {
+                context.replyWarning("Cannot kick user " + targetMember.getUser().getAsTag() + " due to lack of permissions");
+            } catch (HierarchyException e) {
+                context.replyWarning("Cannot kick user " + targetMember.getUser().getAsTag() + " due to them being higher then me");
```
Comment (binaryoverload) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java]: This needs clarifying. Specifically that the user has a role which is higher than the highest role the bot has

File: src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Override
+    public String command() {
+        return "kick";
+    }
+
+    @Override
+    public CascadePermission getPermission() {
+        return CascadePermission.of("Kick Command", "kick", false);
```
Comment (binaryoverload) [inline src/main/java/com/cascadebot/cascadebot/commands/moderation/KickCommand.java]: Add the Permission to this (If you accept this suggestion you'll need to fix the import in intellij)
```suggestion
        return CascadePermission.of("Kick Command", "kick", false, Permission.KICK_MEMBERS);
```

**Pull request merged**